appinventor/appengine/src/com/google/appinventor/client/editor/simple/components/MockContainer.java

```diff
@@ -256,7 +256,8 @@ protected boolean acceptableSource(DragSource source) {
     if (component instanceof MockVisibleComponent) {
       // Sprites are only allowed on Canvas, not other containers.
       // Map features are only allowed on Map, not other containers.
-      if (!(component instanceof MockSprite) && !(component instanceof MockMapFeature)) {
+      // Chart Data components are only allowed on Charts, not other containers.
+      if (!(component instanceof MockSprite) && !(component instanceof MockMapFeature) && !(component instanceof MockChartData)) {
```

Comment:
An additional condition here has been added. Should we possibly think of refactoring this into it's own method, since in the future, more drag & droppable components might be devised, and this condition would become even more complex?